extracted/block_12.py

```diff
@@ -0,0 +1,42 @@
+# readAAsequence
+# Input argument: filename 
+# Takes file with AA sequence, function reads the file, discards the header, returns sequence as a string
+
+# Need to add a way to catch non-fasta files
+
+#Example: readAAsequence("AA.fa")
+
+# Define function(args)
+def readAAsequence(inputfile):
+    
+    # Open file with context manager
+    with open(inputfile, "r") as infile:
+        
+        # Define empty lists for header and sequence values
+        header = []
+        sequence = []
+        
+        first_line = infile.readline().strip()
+        if first_line.startswith(">"):
+            pass
+        else:
+            raise BadSequenceException("""File doesn't seem to be in FASTA format, the FASTA header should start with a >
+            > FASTA HEADER
+            SEQUENCEINFORMATION
+            """)
+        
+        # Find header, sequence
+        for line in infile:
```

Comment:
does this actually work? `infile` is a file pointer not a list 😱

Reply:
also what are you trying to achieve here? Could you just do `infile.read()` and string out the header?